lib/cudadrv/memory.jl

```diff
@@ -62,18 +62,26 @@ Allocate `bytesize` bytes of memory on the device. This memory is only accessibl
 GPU, and requires explicit calls to `unsafe_copyto!`, which wraps `cuMemcpy`,
 for access on the CPU.
 """
-function alloc(::Type{DeviceBuffer}, bytesize::Integer)
+function alloc(::Type{DeviceBuffer}, bytesize::Integer; async::Bool=false)
```

Comment:
Should we make all allocation async by-default?

Reply:
CUDA 11.2-only... I've thought about bumping the minimum requirement, but it might be a bit much.